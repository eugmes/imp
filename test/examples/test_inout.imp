procedure proc (a, b : out integer) is
begin
  a = 42;
  b = 43;
end proc;

procedure proc1 (a, b : in out integer) is
begin
  a = a + 42;
  b = b + 43;
end proc1;

procedure main () is
  var x, y : integer;
begin
  call proc (x, y);
  output(x); newline;
  output(y); newline;
  call proc1 (x, y);
  output(x); newline;
  output(y); newline;
end main;
