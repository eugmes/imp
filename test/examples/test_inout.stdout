42
43
84
86
